--- php-5.2.10/ext/imap/php_imap.c	2009-04-30 13:55:44.000000000 -0500
+++ php5.2-200907161430/ext/imap/php_imap.c	2009-06-23 09:41:06.000000000 -0500
@@ -26,7 +26,7 @@
    | PHP 4.0 updates:  Zeev Suraski <zeev@zend.com>                       |
    +----------------------------------------------------------------------+
  */
-/* $Id: php_imap.c,v 1.208.2.7.2.51 2009/04/30 18:55:44 pajoye Exp $ */
+/* $Id: php_imap.c,v 1.208.2.7.2.53 2009/06/23 12:43:10 iliaa Exp $ */
 
 #define IMAP41
 
@@ -2670,7 +2670,7 @@
 
 	slst = mail_sort(imap_le_struct->imap_stream, (myargc == 6 ? Z_STRVAL_PP(charset) : NIL), spg, mypgm, (myargc >= 4 ? Z_LVAL_PP(flags) : NIL));
 
-	if (spg) {
+	if (spg && myargc >= 4 && !(Z_LVAL_PP(flags) & SE_FREE)) {
 		mail_free_searchpgm(&spg);
 	}
 
@@ -3712,7 +3712,7 @@
 
 	mail_search_full(imap_le_struct->imap_stream, (argc == 4 ? Z_STRVAL_PP(charset) : NIL), pgm, flags);
 
-	if (pgm) {
+	if (pgm && !(flags & SE_FREE)) {
 		mail_free_searchpgm(&pgm);
 	}
 
@@ -4341,7 +4341,7 @@
 
 	pgm = mail_criteria(criteria);
 	top = mail_thread(imap_le_struct->imap_stream, "REFERENCES", NIL, pgm, flags);
-	if (pgm) {
+	if (pgm && !(flags & SE_FREE)) {
 		mail_free_searchpgm(&pgm);
 	}
 
