--- php-5.2.13/ext/gd/libgd/gdft.c.orig	2010-01-08 06:18:52.000000000 -0600
+++ php-5.2.13/ext/gd/libgd/gdft.c	2010-03-24 00:41:32.000000000 -0500
@@ -1101,7 +1101,7 @@
 
 			/* now, draw to our target surface */
 			bm = (FT_BitmapGlyph) image;
-			gdft_draw_bitmap(tc_cache, im, fg, bm->bitmap, x + x1 + ((pen.x + 31) >> 6), y + y1 + ((pen.y + 31) >> 6) - bm->top);
+			gdft_draw_bitmap(tc_cache, im, fg, bm->bitmap, x + x1 + ((pen.x + 31) >> 6) + bm->left, y + y1 + ((pen.y + 31) >> 6) - bm->top);
 		}
 
 		/* record current glyph index for kerning */
