--- php-5.2.8/ext/oci8/config.m4.lib64	2008-05-13 11:06:55.000000000 -0500
+++ php-5.2.8/ext/oci8/config.m4	2008-12-09 21:21:17.000000000 -0600
@@ -371,7 +371,7 @@
   AC_MSG_CHECKING([Oracle Instant Client SDK header directory])
 
 dnl Header directory for Instant Client SDK RPM install
-  OCISDKRPMINC=`echo "$PHP_OCI8_INSTANT_CLIENT" | $PHP_OCI8_SED -e 's!^/usr/lib/oracle/\(.*\)/client\('${LIBDIR_SUFFIX}'\)*/lib[/]*$!/usr/include/oracle/\1/client\2!'`
+  OCISDKRPMINC=`echo "$PHP_OCI8_INSTANT_CLIENT" | $PHP_OCI8_SED -e 's!^/usr/lib\(64\)*/oracle/\(.*\)/client\(.*\)/lib/*$!/usr/include/oracle/\2/client\3!'`
 
 dnl Header directory for Instant Client SDK zip file install
   OCISDKZIPINC=$PHP_OCI8_INSTANT_CLIENT/sdk/include
